src/transformers/models/longt5/modeling_longt5.py

```diff
@@ -0,0 +1,2533 @@
+# coding=utf-8
+# Copyright 2022 Google LLC., LongT5 Authors and HuggingFace Inc. team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" PyTorch LongT5 model."""
+
+
+import copy
+import math
+import warnings
+from typing import List, Optional, Tuple, Union
+
+import torch
+from torch import nn
+from torch.nn import CrossEntropyLoss
+from torch.utils.checkpoint import checkpoint
+
+from ...activations import ACT2FN
+from ...modeling_outputs import (
+    BaseModelOutput,
+    BaseModelOutputWithPastAndCrossAttentions,
+    Seq2SeqLMOutput,
+    Seq2SeqModelOutput,
+)
+from ...modeling_utils import PreTrainedModel, find_pruneable_heads_and_indices, prune_linear_layer
+from ...utils import (
+    DUMMY_INPUTS,
+    DUMMY_MASK,
+    add_start_docstrings,
+    add_start_docstrings_to_model_forward,
+    is_torch_fx_proxy,
+    logging,
+    replace_return_docstrings,
+)
+from ...utils.model_parallel_utils import assert_device_map, get_device_map
+from .configuration_longt5 import LongT5Config
+
+
+logger = logging.get_logger(__name__)
+
+_CONFIG_FOR_DOC = "LongT5Config"
+_TOKENIZER_FOR_DOC = "T5Tokenizer"
+_CHECKPOINT_FOR_DOC = "Stancld/LongT5-Local-Base"
+
+# TODO: Update before the merge
+LONGT5_PRETRAINED_MODEL_ARCHIVE_LIST = [
+    "Stancld/LongT5-Local-Base",
+    "Stancld/LongT5-Local-Large",
+    "Stancld/LongT5-TGlobal-Base",
+    "Stancld/LongT5-TGlobal-Large",
+]
+
+
+####################################################
+# PyTorch Models are constructed by sub-classing
+# - torch.nn.Module for the layers and
+# - PreTrainedModel for the models (it-self a sub-class of nn.Module)
+####################################################
+PARALLELIZE_DOCSTRING = r"""
```

Comment:
Let's remove those - think they don't work well and are out-dated no cc @sgugger @stas00 ?

Reply:
They should work just fine. They are not out dated, since we don't have a better replacement if someone wants pipeline parallelism, this is just very inefficient solution.

Deepspeed works just fine to solve the scalability problem and thus this feature indeed is not needed. the Naive PP was added before we integrated Deepspeed into Transformers.

IMHO, it's totally OK not to replicate the naive parallelism from T5 to these new models.

Reply:
Yes I don't think we need to copy those as we already have a better solution (DeepSpeed) and another one coming soon in Accelerate. Utlimately I think we will deprecate the parallelize API in GPT-2 and T5.